### PR TITLE
Add preprocess-images flag to image-classifier

### PR DESCRIPTION
**Pull request by gcatron**
Summary:
This will preload all images before classifying them, this is useful for benchmarking.
Documentation: NA

Test Plan: Ninja test, and run image-classifier exercising the new option.

### CODE REVIEW

File: tools/loader/ImageClassifier.cpp [hunk 1]
```diff
@@ -457,34 +479,83 @@ int main(int argc, char **argv) {
...
+      llvm::Error runErr = llvm::Error::success();
+      inFlight++;
+      hostManager->runNetwork(
+          name, std::move(batch.second),
+          [&runErr, output, names, &runPromise,
+           &inFlight](runtime::RunIdentifierTy, llvm::Error err,
+                      std::unique_ptr<ExecutionContext> contextPtr) {
+            runErr = std::move(err);
+            // Print the top-k results from the output Softmax tensor.
+            processAndPrintResults(output, names);
```
Comment (nickgg) [inline tools/loader/ImageClassifier.cpp]: The HostManager has several threads, will the results printing now possibly be interleaved on stdout?

File: tools/loader/ImageClassifier.cpp [hunk 2]
```diff
@@ -198,9 +208,8 @@ buildAndCompileAndGetInAndOutPair(Loader &loader, PlaceholderBindings &bindings,
   // Get the Tensor from the Placeholder that the final expected Softmax writes
   // into at the end of image inference.
   Placeholder *SMPH = EXIT_ON_ERR(LD->getSingleOutput());
-  Tensor *SMT = bindings.get(SMPH);
```
Comment (nickgg) [inline tools/loader/ImageClassifier.cpp]: If we're not returning the Tensor do we still need to allocate it (above)?

File: tools/loader/ImageClassifier.cpp [hunk 3]
```diff
@@ -457,34 +479,83 @@ int main(int argc, char **argv) {
...
+            }
+          });
+
+      EXIT_ON_ERR(std::move(runErr));
+    }
+    // wait for all to finish.
+    fut.wait();
+    if (timeOpt) {
+      timer.stopTimer();
+      llvm::outs() << llvm::formatv("Wall time per item (s): {0:f4}\n",
```
Comment (nickgg) [inline tools/loader/ImageClassifier.cpp]: This PR also adds timing/benchmarking functionality to the ImageClassifier? Could you update the summary to reflect this (& how it works, what it looks like, etc)?

File: tools/loader/ImageClassifier.cpp [hunk 4]
```diff
@@ -457,34 +479,83 @@ int main(int argc, char **argv) {
...
+
+      preLoadedImages.push_back(std::move(inputImageData));
+      contexts.emplace(inputImageBatchFilenames, std::move(newContext));
+    } else {
+      loader.runInference(*bindings, batchSize);
+      // Print the top-k results from the output Softmax tensor.
+      numErrors += processAndPrintResults(SMT, inputImageBatchFilenames);
+    }
+  }
+  if (preProcessImages) {
```
Comment (nickgg) [inline tools/loader/ImageClassifier.cpp]: Doesn't this disable running the network if preProcessImages is false?

**Comment by gcatron** [comment]
Had a discussion with @jfix71, Since the goal of this was really to enable benchmarking, and for that we don't really need to use real images I'm going to close this PR and refactor this to more of a pure benchmark use case.

File: tools/loader/ImageClassifier.cpp (hunk 2 above)
Comment (gcatron) [reply tools/loader/ImageClassifier.cpp]: We still need the backing tensor, we just access it via the PH now.

**Pull request closed without merging**